### DISCUSSION TITLE
layui框架的页脚如何彻底关闭(如何彻底取消layui-footer样式)

### DISCUSSION THREAD

**Discussion by Python-tool** [layui/layui#1909] [问答] [answered]
layui框架的页脚如何彻底关闭，框架页面不引入layui-footer样式，底部的区域还是有留白无法去除。
![Uploading 问题截图.png…]()

**Comment by Sight-wcg** [comment]
将 .layui-body 的底部内边距改为 0

**Reply by Python-tool** [reply]
您说的是这个方案吗？也试过了没用，<body class="layui-layout-body close-footer">这个也试了没用：
.layui-body {
    padding-bottom: 0 !important;
}

**Reply by Python-tool** [reply]
class="layui-layout-body close-footer"，这个样式也试过

**Comment by Sight-wcg** [comment, marked as answer]
在大多数现代浏览器中，都内置了强大的开发工具，可用于查看元素样式。以下是一般的步骤：

- 📱 开启浏览器与网页：打开浏览器，并打开要查看元素样式的网页。
- 🔧 打开开发工具的方式： 通常可以通过按 F12 键或在菜单中找到“开发者工具”等选项来打开开发工具面板。
- 🎯 找到相关选项卡： 在开发工具中，找到“元素”或“审查元素”选项卡。
- 🌟 点击元素并高亮显示：然后在网页上点击要查看样式的元素，该元素会在开发工具中高亮显示。
- 📄 查看样式信息： 在开发工具中，可以看到该元素的各种属性，包括样式属性。样式属性会列出该元素所应用的所有 CSS 样式规则，包括内联样式、嵌入样式和外部样式表中的样式。例如，你可能会看到元素的字体大小、颜色、边距、背景等样式信息。通过查看这些样式，你可以更好地理解网页的布局和设计，以及找出样式可能存在的问题或进行样式的调试。
- 🤔 不同浏览器的差异：不同浏览器的开发工具可能会有一些细微的差别，但基本功能和操作是相似的。

**Comment by Python-tool** [comment]
<body class="layui-layout-body">
    <div class="layui-layout layui-layout-admin">
        <div class="layui-header">
        </div>
        <div class="layui-side layui-bg-black">
        </div>
        <div class="layui-body">
            <div class="layui-tab layui-tab-brief" lay-filter="tables" lay-allowclose="true">
                <ul class="layui-tab-title"></ul>
                <div class="layui-tab-content" style="padding: 0px;"></div>
            </div>
        </div>
        
    </div>
    <script src="../static/layui/index.js"></script>
</body>
找到问题所在了：index.js里面的代码有一行根据屏幕高度减去admin模板头部高度（这个减去的高度不准确导致底部页脚存在44PX空白）
    //定义函数 绑定增加tab，删除tab，切换tab几项事件
    var tabFunction = {
        //新增tab url 页面地址 id 对应data-id name标题
        tabAdd: function (url, id, name) {
            _**var height = document.documentElement.clientHeight - 116 + "px";    //解决index页面挂载APP页面时，底部页脚存在44PX空白**_
            element.tabAdd('tables', {
                title: name,
                content: '<iframe data-frameid="' + id + '" scrolling="auto" frameborder="0" src="' + url + '" style="width:100%;height:' + (height) + '"></iframe>',
                id: id
            });
        },
